src/backend/distributed/utils/distribution_column.c

```diff
@@ -129,6 +132,12 @@ BuildDistributionKeyFromColumnName(Relation distributedRelation, char *columnNam
 	Var *distributionColumn = NULL;
 	char *tableName = RelationGetRelationName(distributedRelation);
 
+	/* short circuit for reference tables */
```

Comment:
> I guess the alternative is to not call it for references tables

In general*, I followed the idea that all functions can be called with reference tables. That's why I proposed the change as this here. But, I'm cool with changing if you 

*I had to have an exception for `AddShardIntervalRestrictionToSelect()` which is not possible to call on a reference table and was hard to refactor.